tests/functional/raw-node/test/GCP/bucket/get.js

```diff
@@ -0,0 +1,181 @@
+const assert = require('assert');
+const async = require('async');
+const { GCP } = require('../../../../../../lib/data/external/GCP');
+const { makeGcpRequest } = require('../../../utils/makeRequest');
+const { gcpRequestRetry } = require('../../../utils/gcpUtils');
+const { getRealAwsConfig } =
+    require('../../../../aws-node-sdk/test/support/awsConfig');
+const constants = require('../../../../../../constants');
+
+const credentialOne = 'gcpbackend';
+
+describe('GCP: GET Bucket', function testSuite() {
+    this.timeout(180000);
+    let gcpClient;
+    let config;
+
+    before(() => {
+        config = getRealAwsConfig(credentialOne);
+        gcpClient = new GCP(config);
+    });
+
+    describe('without existing bucket', () => {
+        beforeEach(function beforeFn(done) {
+            this.currentTest.bucketName = `somebucket-${Date.now()}`;
+            return done();
+        });
+
+        it('should return 404 and NoSuchBucket', function testFn(done) {
+            gcpClient.getBucket({
+                Bucket: this.test.bucketName,
+            }, err => {
+                assert(err);
+                assert.strictEqual(err.statusCode, 404);
+                assert.strictEqual(err.code, 'NoSuchBucket');
+                return done();
+            });
+        });
+    });
+
+    describe('with existing bucket', () => {
+        let numberObjects;
```

Comment:
Is this meant to be set as 1 initially?

Reply:
There are before hooks in the inner test suites that will set the `numberObject` variable for the beforeEach hook.